### PR TITLE
New doc on APIs

### PR DESCRIPTION
**Pull request by mpaktiti**
Fixes https://github.com/auth0/docs/issues/2408

### CODE REVIEW

**Comment by auth0build** [comment]
Can one of the admins verify this patch?

File: articles/api/overview.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+# APIs
+
+## Overview
+
+An API is an entity that represents an external resource, capable of accepting and responding to protected resource requests made by clients. At the [OAuth2 spec](https://tools.ietf.org/html/rfc6749) an API maps to the **Resource Server**.
+
+When a client wants to access an API's protected resources it must provide an access token. The same access token can be used to access the API's resources without having to authenticate again, until it expires.
+
+Each access token may include a list of the permissions that have been granted to the client. These permissions are defined using the **scope** request parameter.
```
Comment (cabralmartin) [inline articles/api/overview.md]: Permission are defined by the API. The `scope` parameter allows clients to request a subset of those defined permissions when they execute the authorization flow.

File: articles/api/overview.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+For example, an API that holds a user's appointments, may accept two different levels of authorization: read only (scope `read:appointments`) or write (scope `write:appointments`). When a client asks the API to list a user's appointments, then the access token should contain the `read:appointments` scope. In order to edit an existing appointment or create a new one, the access token should contain the `write:appointments` scope.
+
+> For more information on tokens please refer to: [Tokens used by Auth0](/tokens).
+
+## How to configure an API in Auth0
```
Comment (cabralmartin) [inline articles/api/overview.md]: Just FYI, this subsection will be removed at some point in the future when we make the section available by default.

File: articles/api/overview.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+**NOTE**: Keep in mind that we are working on building quickstarts for more stacks, apart from those currently available.
+
+The other available views for your API are:
+
+- **Settings**: lists the settings for your API. Some are editable. Here you can change the token expiration time and enable offline access (this way Auth0 will allow clients to ask for Refresh Tokens for this API).
+
+- **Scopes**: here you can define the scopes for this API, by setting a name and a description.
+
+- **Non Interactive Clients**: lists your Non Interactive Clients. You can authorize which Clients should use your API and which scopes should be granted to each. Only non-interactive Clients require explicit permission. That is because, when you authorize a non-interactive Client to access an API, Auth0 is creating a Client Grant for that Client. For more details on this case refer to: [Setting up a Client Credentials Grant using the Management Dashboard](/api-auth/config/using-the-auth0-dashboard).
```
Comment (cabralmartin) [inline articles/api/overview.md]: `You can authorize which Clients should use your API and which scopes should be granted to each.`
->
`You can authorize which Non Interactive Clients can request access tokens for your API. You can optionally select a subset of the defined scopes to further limit the access that an authorized client has.`

File: articles/api/overview.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- **RS256** is an [asymmetric algorithm](https://en.wikipedia.org/wiki/Public-key_cryptography) which means that there are two keys: one public and one private (secret). Auth0 has the secret key, which is used to generate the signature, and the consumer of the JWT has the public key, which is used to validate the signature.
+
+- **HS256** is a [symmetric algorithm](https://en.wikipedia.org/wiki/Symmetric-key_algorithm) which means that there is only one secret key, shared between the two parties. The same key is used both to generate the signature and to validate it. Special care should be taken in order for the key to remain confidential.
+
+The most secure practice, and our recommendation, is to use **RS256**. Some of the reasons are:
+
+- With RS256 you are sure that only the holder of the private key (Auth0) can sign tokens, while anyone can check if the token is valid using the public key.
+
+- Under HS256, If the private key is compromised you would have to re-deploy the API with the new secret. With RS256 you can request a token that is valid for multiple audiences.
+- With RS256 you can implement key rotation without having to re-deploy the API with the new secret.
```
Comment (cabralmartin) [inline articles/api/overview.md]: line break

**Pull request merged**